src/ConnectClient/ConnectClient.php

```diff
@@ -19,17 +21,108 @@ class ConnectClient
      */
     public $apiUrl = 'https://connect.givewp.com';
 
+    /**
+     * @unreleased
+     */
+    public function __construct()
+    {
+        if (defined('GIVE_CONNECT_URL') && GIVE_CONNECT_URL) {
+            $this->apiUrl = GIVE_CONNECT_URL;
+        }
+    }
+
     /**
      * Get rest api endpoint url for requests.
      *
-     * @since 2.8.0
+     * @since      2.8.0
+     *
+     * @param string $endpoint The route on the server. E.g: 'gateway-name/connect'
+     */
+    public function getApiUrl(string $endpoint = ''): string
+    {
+        if ( ! empty($endpoint)) {
+            $this->apiUrl .= '/' . ltrim($endpoint, '/');
```

Comment:
I don't see the value in modifying the `apiUrl` property to reflect the endpoint. It's only written and never referenced. In fact, this would have strange behavior:
```php
$client = new ConnectClient();

$client->get('/foo'); // gateway.givewp.com/foo
$client->get('/bar'); // gateawy.givewp.com/foo/bar <-- ???
```

It seems like making this stateless would be more intuitive. It would also mean the class can be stored in the Service Container as a singleton. You'd also be able to pass the connect URL as a constructor, which is passed in via the Container:

```php
give()->singleton(ConnectClient::class, static function() {
    return new ConnectClient(GIVE_CONNECT_URL);
});
```

This would be useful for unit testing.

Reply:
Resolve in eac07fc85e7a099ef0c74b9ba1f5aca9db00ed79 and 7539b3c51978a1131ab745e7eca30325f176dc3a